### PR TITLE
OF-1779: Fix SQL Statement in Pubsub

### PR DESCRIPTION
**Pull request by mightymop**
Fixes Statement for SQL SERVER purge function

### CODE REVIEW

File: xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/PubSubPersistenceManager.java [hunk 1]
```diff
@@ -58,7 +58,7 @@
 
     private static final Logger log = LoggerFactory.getLogger(PubSubPersistenceManager.class);
 
-    private static final String PERSISTENT_NODES = "SELECT serviceID, nodeID, maxItems " +
+    private static final String PERSISTENT_NODES = "SELECT DISTINCT serviceID, nodeID, maxItems " +
```
Comment (guusdk) [inline xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/PubSubPersistenceManager.java]: Although it makes sense that we'd like unique rows to be returned, I'm alarmed by the fact that there's a need to add the DISTINCT keyword. Is there a scenario where the original query returns a non-distinct result set?

**Comment by guusdk** [comment]
Thanks! Exactly what issue does this fix? Can you give a description that we can use to create a new ticket in JIRA please? Afterwards, I'd love for you to re-word the commit message to include the JIRA ticket number.

File: xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/PubSubPersistenceManager.java (hunk 1 above)
Comment (mightymop) [reply xmppserver/src/main/java/org/jivesoftware/openfire/pubsub/PubSubPersistenceManager.java]: I've found doubled rows in database but dont know where they came from... 
this fix indeed would not fix the cause but i will prevent errors based on multiple results...

**Comment by mightymop** [comment]
I think it was OF-1779 (https://issues.igniterealtime.org/browse/OF-1779?jql=text%20~%20%22sql%22)
i also tested all versions of the old statements (they were several commits to that bug)
With this pull request we had no errors anymore...

**Comment by guusdk** [comment]
Awesome, thank you!

**Pull request merged**